### PR TITLE
provide basic bind backend

### PR DESCRIPTION
**Pull request by anthonysomerset**
Hi This PR provides basic Bind backend support

1) will install the correct backend package like other modules
2) the DB parameters are now optional so they are not required for this backend (logic might need to be tightened up in other backends to check and "require" them.)
3) still need to set `backend_install` and `backend_create_tables` to false (yet to make these parameters optional)

I am working on a seperate resource type to "create" zones at the very least for bind will open a separate PR once i have completed this more.

### CODE REVIEW

**Comment by ju5t** [comment]
I've changed the merge request to target the develop branch. 

Let me know when you're ready for a review.

**Comment by anthonysomerset** [comment]
I'm just about ready to review - i had fixed the initial build and formatting errors and introduced some basic tests for the new backend too - theres a test error on puppet 5 now but i'm not 100% sure how the test can be improved accordingly because the parameter is now optional to support bind backend

**Comment by ju5t** [comment]
The test expects the error message to contain:

```
parameter 'db_username' expects a String\[1, default\] value, got String
```

But the error message is:

```
Error while evaluating a Resource Statement, Class[Powerdns]: parameter 'db_username' expects a value of type Undef or String[1, default], got String
```

The test should be:

```
expect { subject.call } .to raise_error(/parameter 'db_username' expects a value of type Undef or String\[1, default\], got String/)
```

Instead of

```
expect { subject.call } .to raise_error(/parameter 'db_username' expects a String\[1, default\] value, got String/)
```

File: manifests/backends/bind.pp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  }
+
+  file { "${::powerdns::params::authoritative_configdir}/bind":
+    ensure => directory,
+    mode   => '0755',
+    owner  => 'root',
+    group  => 'root',
+  }
+
+  file_line { 'powerdns-bind-baseconfig':
```
Comment (ju5t) [inline manifests/backends/bind.pp]: This resource is not tested yet. Coverage dropped to 98.53%.

Comment (anthonysomerset) [reply manifests/backends/bind.pp]: this item should now be fixed

**Comment by anthonysomerset** [comment]
Test coverage updated - now covering 100% of resources

Slightly annoyingly - that DB test fix *only* works on Puppet 5, this test now produces errors on puppet versions < 5

i'm not sure we want to bump minimum puppet version to 5 over a test - but not sure how to make this test work between differing puppet versions accordingly

**Comment by ju5t** [comment]
I didn't expect that.

You could test per Puppet version too but it seems a bit much. It's easier to improve the regex:

```
expect { subject.call } .to raise_error(/parameter 'db_username' expects a (value of type Undef or )?String\[1, default\]( value)?, got String/)
```

I've tested this on Puppet 4.7 and 5.

**Comment by anthonysomerset** [comment]
I've just made these final changes - tweaking grammer and fixing the test cases

have also changed names to named.conf to better align with manuals and /etc/pdns/named

**Comment by ju5t** [comment]
@anthonysomerset I have nothing else to add. If you want I can merge this into our develop branch.

**Comment by anthonysomerset** [comment]
yes please - i've got nothing more on this at this stage, thanks

**Comment by ju5t** [comment]
Thanks for the new feature. I'll release 1.3.1 later this week.

**Pull request merged**